@here/harp-mapview/lib/image/MapViewImageCache.ts

```diff
@@ -83,10 +82,56 @@ export class MapViewImageCache {
         if (startLoading === true) {
             return ImageCache.instance.loadImage(imageItem);
         }
-
         return imageItem;
     }
 
+    /**
+     * Remove the image with this name from the cache.
+     *
+     * @param name - Name of the image.
+     * @returns `true` if item has been removed.
+     */
+    removeImage(name: string): boolean {
+        return this.removeImageInternal(name);
+    }
+
+    /**
+     * Remove images using the URL from the cache.
+     *
+     * @param url - URL of the image.
+     * @returns `true` if image has been removed. If multiple images are referring to the same
+     * image URL, they are all removed.
+     */
+    removeImageByUrl(url: string): boolean {
+        const names = this.m_url2Name.get(url);
+        if (names !== undefined) {
+            for (const name of [...names]) {
```

Comment:
why do you need to wrap `names` in `[... ]` ?

Reply:
It copies the content of names into a temporary list, which is probably a good thing because we are working on the original list, no?

Reply:
Good point! Ok, please add a comment and / or test case that will fail if some smart person tries to "fix this''

Reply:
Can do. But I got the nagging feeling that we have to comment a lot of code that uses tricky stuff.